### PR TITLE
[SYCLomatic] Fix incorrect migration of thrust::stable_sort called in template function and enable migration of thrust::not2

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/CallExprRewriterThrust.cpp [hunk 1]
```diff
@@ -21,7 +21,30 @@ class CheckThrustArgType {
   bool operator()(const CallExpr *C) {
     std::string ArgType;
     int NumArgs = C->getNumArgs();
+
     if (Idx < NumArgs) {
+      // template<typename ELT_TYPE>
```
Comment (AndyCHHuang) [inline clang/lib/DPCT/CallExprRewriterThrust.cpp]: I think the code example could be simplified.
// template\<typename T\>      
// void testfunc() {
// thrust::host_vector\<T\> V(1);
//  thrust::stable_sort(V.begin(),V.end(),thrust::not2(thrust::greater_equal()));
// }

Comment (tomflinda) [reply clang/lib/DPCT/CallExprRewriterThrust.cpp]: okay

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/CallExprRewriterThrust.cpp [hunk 2]
```diff
@@ -21,7 +21,29 @@ class CheckThrustArgType {
...
     if (Idx < NumArgs) {
+      // template<typename T>
+      // void testfunc() {
+      //  thrust::host_vector<T> V(1);
+      //  thrust::stable_sort(V.begin(),V.end(),thrust::not2(thrust::greater_equal()));
+      // }
+      // void foo() {
+      //   testfunc<int>();
+      // }
+      // For the code above argument "A.begin()" has type
```
Comment (zhimingwang36) [inline clang/lib/DPCT/CallExprRewriterThrust.cpp]: A.begin ==> V.begin

Comment (tomflinda) [reply clang/lib/DPCT/CallExprRewriterThrust.cpp]: ok

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**